source/framework/sensitivity/src/TRestComponent.cxx

```diff
@@ -292,23 +292,20 @@ ROOT::RDF::RNode TRestComponent::GetMonteCarloDataFrame(Int_t N) {
     ROOT::RDF::RNode df = ROOT::RDataFrame(N);
 
     // Function to fill the RDataFrame using GetRandom method
-    auto fillDataFrame = [this]() {
+   auto fillRndm = [this]() {
         auto randomValues = GetRandom();
+        return randomValues;
+   }
+   df.Define("Rndm", fillRndm)
 
-        // Check if the size of randomValues matches the size of columnNames
-        if (randomValues.size() != fVariables.size()) {
-            throw std::runtime_error("Mismatch in sizes of fVariables and randomValues");
-        }
-    };
-
-    // Fill the RDataFrame with values from GetRandom
     for (size_t i = 0; i < fVariables.size(); ++i) {
-        df = df.Define(fVariables[i], [&randomValues, &i]() { return randomValues[i]; });
+        auto varName = fVariables[i];
+        auto FillRand = [&i = i]( const ROOT::RVec<double> randomValues){
+            return randomValues[i];
+        }
+        df = df.Define(varName,  FillRand, "Rndm");
```

Comment:
Then, when trying to compile I get errors in this line. It complains about the arguments.

```
/afs/desy.de/user/j/jgalan/rest-framework/source/framework/sensitivity/src/TRestComponent.cxx:305:46: error: no matching function for call to ‘ROOT::RDF::RInterface<ROOT::Detail::RDF::RNodeBase, void>::Define(std::basic_string<char>&, TRestComponent::GetMonteCarloDataFrame(Int_t)::<lambda(ROOT::VecOps::RVec<double>)>&, const char [5])’
  305 |      df = df.Define(varName, FillRand, "Rndm");
```